src/map/status.cpp

```diff
@@ -544,6 +544,363 @@ uint64 SizeFixDatabase::parseBodyNode(const ryml::NodeRef& node) {
+
+				if( !this->asBool( gradeNode, "Announce", announce ) ){
+					return 0;
+				}
+
+				grade->announce = announce;
+			}else{
+				if( !gradeExists ){
+					grade->announce = false;
+				}
+			}
+
+			if( this->nodeExists( gradeNode, "Catalysator") ){
+				const ryml::NodeRef& catalysatorNode = gradeNode["Catalysator"];
+
+				if( this->nodeExists( catalysatorNode, "Item" ) ){
+					std::string itemName;
+
+					if( !this->asString( catalysatorNode, "Item", itemName ) ){
+						return 0;
+					}
+
+					std::shared_ptr<item_data> id = item_db.search_aegisname( itemName.c_str() );
+
+					if( id == nullptr ){
+						this->invalidWarning( catalysatorNode["Item"], "Unknown item \"%s\".\n", itemName.c_str() );
+						return 0;
+					}
+
+					grade->catalysator.item = id->nameid;
+				}else{
+					if( !gradeExists ){
+						grade->catalysator.item = 0;
+					}
+				}
+
+				if( this->nodeExists( catalysatorNode, "AmountPerStep" ) ){
+					uint16 amountPerStep;
+
+					if( !this->asUInt16( catalysatorNode, "AmountPerStep", amountPerStep ) ){
+						return 0;
+					}
+
+					grade->catalysator.amountPerStep = amountPerStep;
+				}else{
+					if( !gradeExists ){
+						grade->catalysator.amountPerStep = 0;
+					}
+				}
+
+				if( this->nodeExists( catalysatorNode, "MaximumSteps" ) ){
+					uint16 maximumSteps;
+
+					if( !this->asUInt16( catalysatorNode, "MaximumSteps", maximumSteps ) ){
+						return 0;
+					}
+
+					grade->catalysator.maximumSteps = maximumSteps;
+				}else{
+					if( !gradeExists ){
+						grade->catalysator.maximumSteps = 0;
+					}
+				}
+
+				if( this->nodeExists( catalysatorNode, "ChanceIncrease" ) ){
+					uint16 chanceIncrease;
+
+					if( !this->asUInt16Rate( catalysatorNode, "ChanceIncrease", chanceIncrease ) ){
+						return 0;
+					}
+
+					grade->catalysator.chanceIncrease = chanceIncrease;
+				}else{
+					if( !gradeExists ){
+						grade->catalysator.chanceIncrease = 0;
+					}
+				}
+			}else{
+				if( !gradeExists ){
+					grade->catalysator.item = 0;
+					grade->catalysator.amountPerStep = 0;
+					grade->catalysator.maximumSteps = 0;
+					grade->catalysator.chanceIncrease = 0;
+				}
+			}
+
+			if( this->nodeExists( gradeNode, "Options" ) ){
+				for( const ryml::NodeRef& optionNode : gradeNode["Options"] ){
+					uint16 optionIndex;
+
+					if( !this->asUInt16( optionNode, "Option", optionIndex ) ){
+						return 0;
+					}
+
+					std::shared_ptr<s_enchantgradeoption> option = util::map_find( grade->options, optionIndex );
+					bool optionExists = option != nullptr;
+
+					if( !optionExists ){
+						option = std::make_shared<s_enchantgradeoption>();
+						option->id = optionIndex;
+					}
+
+					if( this->nodeExists( optionNode, "Item" ) ){
+						std::string itemName;
+
+						if( !this->asString( optionNode, "Item", itemName ) ){
+							return 0;
+						}
+
+						std::shared_ptr<item_data> id = item_db.search_aegisname( itemName.c_str() );
+
+						if( id == nullptr ){
```

Comment:
How to remove an existing entry on import and go back to the default value `option->item = 0;` ?